bcgov/cas-registration
# Enhancement/breadcrumbs/223

Addresses [Ticket 223](https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/223)

## 🚧 Changes

    - Update client\app\(authenticated)\dashboard\select-operator\request-access to route group syntax: client\app\(authenticated)\dashboard\select-operator\(request-access)
    - Update client\app\components\navigation\Bread with fn isValidLink

## 🚀 Impact:
   - In the app directory, nested folders are normally mapped to URL paths. However, you can mark a folder as a Route Group to prevent the folder from being included in the route's URL path. This allows you to organize your route segments and project files into logical groups without affecting/**displaying** the URL path structure.
   - isValidLink introduces an ignore list that will not display in the breadcrumbs (e.g. confirm; received)


##  🔬 Testing:

1. Open a terminal in cas-registration\bc_obps directory
2. Follow the [developer environment setup](https://github.com/bcgov/cas-registration/blob/develop/docs/developer-environment-setup.md#backend-environment-setup) to start the API
3.  Open a terminal in cas-registration\client folder
4. Start the development server:
   ```
   yarn dev
   ```
5. In your web browser, navigate to `http://localhost:3000/dashboard/select-operator`
6. Select an operator and click "Request Access"
7. Validate the "confirm" screen displays expected content without breadcrumbs "Request Access" and "Confirm" 
8. Test other "Request Access" functionalities...
9. ...screen displays expected content without breadcrumbs expected to be ignored


![image](https://github.com/bcgov/cas-registration/assets/120038448/06c8016d-332f-4472-bdc8-bebf5d2b376a)
